converter.go

```diff
@@ -1634,96 +1637,306 @@ func buildListFromNativeArrow(ctx context.Context, rowIdx int, fieldMetadata fie
+					if resInt == nil {
+						return sql.NullByte{Valid: false}, nil
+					}
+					return sql.NullByte{Valid: true, Byte: resInt.(byte)}, nil
+				})
+			}
 			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (int8, error) {
-				return arrowInt8ToValue(typedValues, j, higherPrecision, fieldMetadata.Scale).(int8), nil
+				resInt := arrowInt8ToValue(typedValues, j, higherPrecision, fieldMetadata.Scale)
+				if resInt == nil {
+					return 0, errNullValueInArray()
+				}
+				return resInt.(int8), nil
 			})
 		}
 	case realType:
+		if arrayValuesNullableEnabled(ctx) {
+			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (sql.NullFloat64, error) {
+				resFloat := arrowRealToValue(values.(*array.Float64), j)
+				if resFloat == nil {
+					return sql.NullFloat64{Valid: false}, nil
+				}
+				return sql.NullFloat64{Valid: true, Float64: resFloat.(float64)}, nil
+			})
+		}
 		return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (float64, error) {
-			return arrowRealToValue(values.(*array.Float64), j).(float64), nil
+			resFloat := arrowRealToValue(values.(*array.Float64), j)
+			if resFloat == nil {
+				return 0, errNullValueInArray()
+			}
+			return resFloat.(float64), nil
 		})
 	case textType:
+		if arrayValuesNullableEnabled(ctx) {
+			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (sql.NullString, error) {
+				resString := arrowStringToValue(values.(*array.String), j)
+				if resString == nil {
+					return sql.NullString{Valid: false}, nil
+				}
+				return sql.NullString{Valid: true, String: resString.(string)}, nil
+			})
+		}
 		return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (string, error) {
-			return arrowStringToValue(values.(*array.String), j).(string), nil
+			resString := arrowStringToValue(values.(*array.String), j)
+			if resString == nil {
+				return "", errNullValueInArray()
+			}
+			return resString.(string), nil
 		})
 	case booleanType:
+		if arrayValuesNullableEnabled(ctx) {
+			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (sql.NullBool, error) {
+				resBool := arrowBoolToValue(values.(*array.Boolean), j)
+				if resBool == nil {
+					return sql.NullBool{Valid: false}, nil
+				}
+				return sql.NullBool{Valid: true, Bool: resBool.(bool)}, nil
+			})
+		}
 		return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (bool, error) {
-			return arrowBoolToValue(values.(*array.Boolean), j).(bool), nil
+			resBool := arrowBoolToValue(values.(*array.Boolean), j)
+			if resBool == nil {
+				return false, errNullValueInArray()
+			}
+			return resBool.(bool), nil
+
 		})
+
 	case binaryType:
 		return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) ([]byte, error) {
-			return arrowBinaryToValue(values.(*array.Binary), j).([]byte), nil
+			res := arrowBinaryToValue(values.(*array.Binary), j)
+			if res == nil {
```

Comment:
Needed? Can be inlined?

Reply:
Needed because of the panic with assertion we disciussed. Can't inline it, because it's used in the "else" statement that can't be a proper else because of linter